### PR TITLE
Dfr 3194 document removal submitted

### PR DESCRIPTION
**Pull request by ptrelease**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/DFR-3194

### Change description ###

Add document removal. 

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentremoval/DocumentRemovalService.java [hunk 1]
```diff
@@ -49,4 +162,89 @@ public void retrieveDocumentNodes(JsonNode root, List<JsonNode> documentNodes) {
...
+                genericDocumentService.deleteDocument(documentToRemove.getCaseDocument().getDocumentUrl(), authorisationToken);
+            }
+        } catch (Exception e) {
+            log.error(format(
+                    "Failed to delete document url %s",
+                    documentToRemove.getCaseDocument().getDocumentUrl()), e);
+
+            throw new DocumentDeleteException(e.getMessage(), e);
+        }
+    }
```
Comment (ptrelease) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentremoval/DocumentRemovalService.java:237]: Ideally leave this comment.  The async approach hasn't been tested in demo, but the the sync approach has.
When isSecureDocEnabled is toggled on, files will begin deleting upon file removal.  There would be an option to make file deletion async then for performance.

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/documentremoval/DocumentRemovalAboutToSubmitHandler.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.documentremoval.DocumentRemovalService;
+
+
+@Slf4j
+@Service
+public class DocumentRemovalAboutToSubmitHandler extends FinremCallbackHandler {
+
+    private final DocumentRemovalService documentRemovalService;
+
+    @Autowired
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/documentremoval/DocumentRemovalAboutToSubmitHandler.java:24]: @Autowired is no longer required where there is only one constructor.
https://docs.spring.io/spring-framework/reference/core/beans/annotation-config/autowired.html

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentremoval/DocumentRemovalService.java [hunk 3]
```diff
@@ -3,33 +3,146 @@
...
+
+    /**
+     * We expect the upload timestamp to be null or valid. However, if anything unexpected is provided,
+     * catch and return null - this is only for sorting on the page.
+     *
+     * @param documentNode a JsonNode containing the upload timestamp
+     * @return documentNodeUploadTimestamp a localDateTime version of the upload timestamp
+     */
+    public LocalDateTime getUploadTimestampFromDocumentNode(JsonNode documentNode) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentremoval/DocumentRemovalService.java:97]: Can be private

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentremoval/DocumentRemovalService.java [hunk 4]
```diff
@@ -3,33 +3,146 @@
...
+        LocalDateTime documentNodeUploadTimestamp;
+        try {
+            documentNodeUploadTimestamp =
+                    Objects.isNull(documentNode.get(DOCUMENT_UPLOAD_TIMESTAMP)) ? null :
+                            LocalDateTime.parse(documentNode.get(DOCUMENT_UPLOAD_TIMESTAMP).asText());
+        } catch (Exception e) {
+            log.info(format(
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentremoval/DocumentRemovalService.java:104]: Would log.error be more appropriate than log.info?

**Review by dawudgovuk — APPROVED** [review]

**Review by jthmcts — APPROVED** [review]

**Pull request merged**